script.js

```diff
@@ -1,12 +1,363 @@
+    (episode) =>
+      episode.name.toLowerCase().includes(searchField.value.toLowerCase()) ||
+      episode.summary.toLowerCase().includes(searchField.value.toLowerCase())
+  );
+
+  if (searchField.value.length > 0) {
+    displayingFilterResults.textContent = `Found ${episodeFilter.length} episodes`;
+    removeCardLayout();
+    clearCards();
+    makePageForEpisodes(episodeFilter);
+  } else {
+    searchField.removeEventListener("input", searchFilterEpisodes);
+    searchField.addEventListener("input", searchFilterShows);
+    showDropDown.value = "default";
+    removeCardLayout();
+    displayingFilterResults.textContent = "";
+    clearCards();
+    render();
+  }
+
+  const searchLayout = document.querySelector(".search__layout");
+  searchLayout.appendChild(displayingFilterResults);
+}
+
+async function makePageForShows(episodeList) {
+  const layoutSelector = document.querySelector(".layout");
+  layoutSelector.innerHTML = "";
+  // Creates a card for each episode
+  episodeList.forEach((episode) => {
+    const filmCard = createShowCard(episode);
+    layoutSelector.appendChild(filmCard);
+
+    filmCard.addEventListener("click", async function () {
+      searchField.value = "";
+      searchField.setAttribute("placeholder", "Search for episode/s...");
+      displayingFilterResults.textContent = "";
+      currentState.showsID = episode.id;
+      currentState.showSelected = episode;
+      const episodeData = await getAllEpisodes();
+      const selectedShow = [...showDropDown].find(
+        (show) => show.value === episode.name
+      );
+      selectedShow.selected = true;
+      currentState.episodesFetched = false;
+      scrollTo(top);
+      addCardLayout();
+      clearCards();
+      makePageForEpisodes(episodeData);
+      searchField.removeEventListener("input", searchFilterShows);
+      searchField.addEventListener("input", searchFilterEpisodes);
+    });
+  });
+}
+
+async function makePageForEpisodes(episodeCard) {
+  const layoutSelector = document.querySelector(".layout");
+  addCardLayout();
+  layoutSelector.innerHTML = "";
+  episodeCard.forEach((episode) => {
+    const filmCard = createEpisodeCard(episode);
+    layoutSelector.appendChild(filmCard);
+  });
+}
+
+// Creates footer
+function createFooter() {
+  const footerElement = createElement("footer", "footer");
+  const layoutSelector = document.querySelector(".layout");
+
+  footerElement.innerHTML = `Data originally sourced by
+        <a href="https://www.tvmaze.com/" id="tv-maze-link">TVMaze.com</a>`;
+  layoutSelector.insertAdjacentElement("afterend", footerElement);
+}
+
+function clearCards() {
+  const layoutSelector = document.querySelector(".layout");
+  layoutSelector.innerHTML = "";
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+async function render() {
+  try {
+    await getAllShows().then((allEpisodes) => {
```

Comment:
Are we working with a list of shows or with a list of episodes in this part of the code?